cmd/thanos/sidecar.go

```diff
@@ -298,13 +322,39 @@ func runSidecar(
 	return nil
 }
 
+func validatePrometheus(ctx context.Context, logger log.Logger, m *promMetadata) error {
+	if m.version == nil {
+		level.Warn(logger).Log("msg", "fetched version is nil or invalid, skip validation")
+		return nil
+	}
+
+	if m.version.LessThan(promclient.FlagsVersion) {
+		level.Warn(logger).Log("msg",
+			"Prometheus doesn't support flags endpoint, skip validation", "version", m.version.Original())
+		return nil
+	}
+
+	flags, err := promclient.ConfiguredFlags(ctx, logger, m.promURL)
+	if err != nil {
+		return errors.Wrap(err, "failed to check flags")
+	}
+	// Check if min-block-time and max-block-time are 2h.
+	if flags.TSDBMinTime != model.Duration(2*time.Hour) || flags.TSDBMaxTime != model.Duration(2*time.Hour) {
```

Comment:
So.. there are people who still want to run non 2h blocks. I think we should:

* Check IF compaction is disabled by checking `if flags.TSDBMinTime !=  flags.TSDBMaxTime` and fail if true
* Check if MinTime == 2h and produce WARNING only. 

What do you think? With that approach we are safe and we warned people. Also this is what this error line suggests. Disable compaction is must have, 2h is only recommended.

cc @tarrall

Reply:
Fair point.